README.md
# TYPO3 Extension devtools

[![Latest Stable Version](https://img.shields.io/packagist/v/ichhabrecht/devtools.svg)](https://packagist.org/packages/ichhabrecht/devtools)
[![Build Status](https://img.shields.io/travis/IchHabRecht/devtools/master.svg)](https://travis-ci.org/IchHabRecht/devtools)
[![StyleCI](https://styleci.io/repos/22914124/shield?branch=master)](https://styleci.io/repos/22914124)

Show modified files from extensions and update the value of `_md5_values_when_last_written` in ext_emconf.php.


ChangeLog
2018-10-03         [RELEASE] Release of devtools 1.1.0 (Nicole Cordes)
2018-10-03 4a12844 [TASK] Raise TYPO3 dependencies to include 9.5 (Nicole Cordes)
2018-10-03 dde6a97 [TASK] Update supporting files to EXT:my_skeleton 0.1.5 (Nicole Cordes)
2018-10-03 9c2caff [BUGFIX] Add missing PHP requirement (Nicole Cordes)
2018-10-03 d68306e [BUGFIX] Obtain severity class from proper object (Nicole Cordes)
2018-10-03 4c8cfdc [BUGFIX] Use AjaxRoutes instead of AjaxHandler (Nicole Cordes)
2018-05-27 0210f13 [FEATURE] add cli support for EM_CONF update (#4) (sventendo)

2018-03-13         [RELEASE] Release of devtools 1.0.0 (Nicole Cordes)
2018-03-13 7cc2094 [FEATURE] Add new extension icon (Nicole Cordes)
2018-03-13 afb9f44 [TASK] Raise compatibility to TYPO3 8.7 (Nicole Cordes)
2018-03-13 001ae33 [BUGFIX] Remove code complexity according to SonarQube Scanner (Nicole Cordes)
2018-03-13 e196c7e [TASK] Update Travis CI configuration (Nicole Cordes)
2018-03-13 1fce72e [FEATURE] Move JavaScript to AMD module (Nicole Cordes)
2018-03-13 3488797 [CLEANUP] Import classes (Nicole Cordes)
2018-03-13 0bc68f1 [TASK] Update coding style rules (Nicole Cordes)
2018-03-13 a16d8eb [TASK] Fix coding style according to php-cs-fixer (Nicole Cordes)
2018-03-13 cd0b009 [TASK] Move js file to JavaScript folder (Nicole Cordes)
2017-03-05 7a36a57 [TASk] Move friendsofphp/php-cs-fixer to suggest section (Nicole Cordes)
2017-03-05 f34e9c8 [TASK] Add more coding style rules (Nicole Cordes)
2017-02-21 1a49eaa [FEATURE] Add PHP lint for supported PHP versions (Nicole Cordes)
2017-02-21 7dfc954 [TASK] Fix coding style according to php-cs-fixer (Nicole Cordes)
2017-02-21 44af42f [FEATURE] Add StyleCI Coding Style Service (Nicole Cordes)
2017-02-21 e0a05ef [FEATURE] Add friendsofphp/php-cs-fixer as dev requirement (Nicole Cordes)
2017-02-21 d5c69c2 [TASK] Exclude local .php_cs files in .gitignore (Nicole Cordes)

2017-02-20         [RELEASE] Release of devtools 0.2.4 (Nicole Cordes)
2017-02-20 8ee62e2 [TASK] Raise compatibility to TYPO3 8.6 (Nicole Cordes)

2016-12-22         [RELEASE] Release of devtools 0.2.3 (Nicole Cordes)
2016-12-22 902e7e2 [TASK] Raise compatibility to TYPO3 8.5 (Nicole Cordes)
2016-12-22 080aa8b [BUGFIX] Ensure packageManager knows all available packages (Nicole Cordes)

2016-11-08         [RELEASE] Release of devtools 0.2.2 (Nicole Cordes)
2016-11-08 b06a42c [TASK] Add composer.json (Nicole Cordes)
2016-11-08 bb654b3 [TASK] Raise version compatibility to TYPO3 8.4.99 (Nicole Cordes)

2016-03-16         [RELEASE] Release of devtools 0.2.1 (Nicole Cordes)
2016-03-16 10f54ca [BUGFIX] Remove removed JavaScript functions (Nicole Cordes)

2015-10-29         [RELEASE] Release of devtools 0.2.0 (Nicole Cordes)
2015-10-29 93ecdb7 [TASK] Set dependency to TYPO3 7.5 (Nicole Cordes)
2015-10-29 568080f [TASK] Reformat code according to TYPO3 Coding Guidelines (Nicole Cordes)
2015-10-29 aec6a72 [TASK] Use JS Modal window Api (Nicole Cordes)
2015-10-29 e19c68d [TASK] Remove unnecessary array_filter function (Nicole Cordes)
2015-10-29 062b48a [BUGFIX] Fix initialization of TYPO3\CMS\Extensionmanager\Utility\EmConfUtility (Nicole Cordes)
2015-10-29 fc2204c [TASK] Use new Icon Api (Nicole Cordes)
2015-10-29 3450cca [TASK] Streamline ProcessActions access point (Nicole Cordes)
2015-10-29 4087102 [TASK] Replace old PackageManager (Nicole Cordes)

2014-11-12         [RELEASE] Release of devtools 0.1.1 (Nicole Cordes)
2014-11-11 3c3f95e [BUGFIX] Fix referenced parameter for PHP 5.4 and above (Nicole Cordes)
2014-09-30 0e17eb5 [BUGFIX] Prevent empty values in file arrays (Nicole Cordes)
2014-09-30 5a2d1d3 [BUGFIX] Force hash value to be an array (Nicole Cordes)

2014-08-14         [RELEASE] Release of devtools 0.1.0 (Nicole Cordes)
2014-08-14 6959991 [TASK] Added documentation (Nicole Cordes)
2014-08-14 5365d48 [FEATURE] Add action to rewrite ext_emconf.php file for modified extensions (Nicole Cordes)
2014-08-14 0b16a0b [FEATURE] Display error dialog if an error occurs (Nicole Cordes)
2014-08-13 2f795d2 [TASK] Add translations (Nicole Cordes)
2014-08-13 bb0b5e8 [TASK] Rework file and folder structure (Nicole Cordes)
2014-08-13 ed42003 [BUGFIX] Add javascript action to dataTables draw listener (Nicole Cordes)
2014-08-13 81c8448 [FEATURE] Add action to display modified files by md5 file check (Nicole Cordes)
2014-08-06 998efd1 [TASK] Add basis files (Nicole Cordes)


ext_emconf.php
<?php

/***************************************************************
 * Extension Manager/Repository config file for ext "devtools".
 *
 * Auto generated 03-10-2018 23:22
 *
 * Manual updates:
 * Only the data in the array - everything else is removed by next
 * writing. "version" and "dependencies" must not be touched!
 ***************************************************************/

$EM_CONF[$_EXTKEY] = array (
  'title' => 'Developer tools',
  'description' => 'Adds nice developer tools to your TYPO3 installation',
  'category' => 'be',
  'state' => 'stable',
  'uploadfolder' => 0,
  'createDirs' => '',
  'clearCacheOnLoad' => 0,
  'author' => 'Nicole Cordes',
  'author_email' => 'typo3@cordes.co',
  'author_company' => 'biz-design',
  'version' => '1.1.0',
  'constraints' => 
  array (
    'depends' => 
    array (
      'php' => '5.6.0-7.2.99',
      'typo3' => '7.6.0-9.5.99',
    ),
    'conflicts' => 
    array (
    ),
    'suggests' => 
    array (
    ),
  ),
  '_md5_values_when_last_written' => 'a:25:{s:9:"ChangeLog";s:4:"fb1d";s:7:"LICENSE";s:4:"b234";s:9:"README.md";s:4:"9b60";s:13:"composer.json";s:4:"c0e1";s:12:"ext_icon.png";s:4:"a925";s:17:"ext_localconf.php";s:4:"9d74";s:16:"phpunit.xml.dist";s:4:"7d0d";s:24:"sonar-project.properties";s:4:"ee73";s:59:"Classes/Command/ExtensionConfigurationCommandController.php";s:4:"b009";s:50:"Classes/Controller/Slot/AbstractSlotController.php";s:4:"3c7d";s:83:"Classes/Controller/Slot/Extensionmanager/ProcessActions/ModifiedFilesController.php";s:4:"1b83";s:93:"Classes/Controller/Slot/Extensionmanager/ProcessActions/UpdateConfigurationFileController.php";s:4:"90a8";s:48:"Classes/Slot/Extensionmanager/ProcessActions.php";s:4:"8a66";s:36:"Classes/Utility/ExtensionUtility.php";s:4:"580e";s:36:"Configuration/Backend/AjaxRoutes.php";s:4:"8e5c";s:26:"Documentation/Includes.txt";s:4:"6d5f";s:23:"Documentation/Index.rst";s:4:"5363";s:26:"Documentation/Settings.yml";s:4:"03e2";s:46:"Documentation/Images/dialog_modified_files.jpg";s:4:"b064";s:40:"Documentation/Images/list_view_icons.jpg";s:4:"340f";s:36:"Documentation/Introduction/Index.rst";s:4:"a10c";s:43:"Resources/Private/Language/de.locallang.xlf";s:4:"c6fe";s:40:"Resources/Private/Language/locallang.xlf";s:4:"dfe3";s:36:"Resources/Public/Icons/Extension.png";s:4:"a925";s:39:"Resources/Public/JavaScript/Devtools.js";s:4:"140a";}',
);

